### PR TITLE
Add library pages for boost,gml,trilinos

### PR DESCRIPTION
**Pull request by kevinstratford**
Now available as modules

### CODE REVIEW

File: docs/software-libraries/boost.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    The following libraries are installed: `atomic chrono container context
+    contract coroutine date_time exception fiber filesystem graph_parallel
+    graph iostreams locale log math mpi program_options random
+    regex serialization stacktrace system test thread timer type_erasure
+    wave`
+
+## Compiling Boost
+
+The Arcer2 github repository contains a recipe for compiling Boost for
```
Comment (aturner-epcc) [inline docs/software-libraries/boost.md:40]: Typo, should be ARCHER2

File: docs/software-libraries/boost.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+the different programming environments.
+```
+$ git clone https://github.com/ARCHER2-HPC/pe-scripts.git
+$ cd pe-scripts
+$ git checkout cse-develop
+$ ./sh/boost.sh --prefix=/path/to/install/location
+```
+where the `--prefix` determines the install location. The list of
+libraries compiled is specified in the `boost.sh` script. See the
+Archer2 [github repository][4] for further information.
```
Comment (aturner-epcc) [inline docs/software-libraries/boost.md:50]: Typo  should be ARCHER2

File: docs/software-libraries/glm.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+### Version history
+
+- Module `glm/0.9.9.6` installed January 2021
+
+
+## Install your own version
+
+One can follow the instructions used to install the current version
+on Archer2 via the Archer2 github repository:
```
Comment (aturner-epcc) [inline docs/software-libraries/glm.md:28]: Typo, should be ARCHER2

File: docs/software-libraries/glm.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```
+$ git clone https://github.com/ARCHER2-HPC/pe-scripts.git
+$ cd pe-scripts
+$ git checkout cse-develop
+$ ./sh/glm.sh --prefix=/path/to/install/location
+```
+where the `--prefix` option sets the install location. See the Archer2
```
Comment (aturner-epcc) [inline docs/software-libraries/glm.md:35]: Typo, should be ARCHER2

File: docs/software-libraries/trilinos.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    Tpetra TrilinosCouplings Triutils Xpetra Zoltan Zoltan2`
+
+    Known issue: Trilinos is not available in `PrgEnv-aocc` at the moment.
+
+    Known issue: The `ForTrilinos` package is not available in this version.
+
+## Compiling Trilinos
+
+A script which has details of the relevant configuration options
+for Trilinos is available at the Archer2 repository. The script
```
Comment (aturner-epcc) [inline docs/software-libraries/trilinos.md:43]: Typo, should be ARCHER2

File: docs/software-libraries/trilinos.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+$ ...
+$ ./sh/trilinos.sh --prefix=/path/to/install/location
+```
+where `--prefix` sets the installation location. The ellipsis `...`
+is standing for the dependencies used to build Trilinos, which
+here are: `metis, parmetis, superlu, superlu-dist, scotch, mumps,
+glm, boost`. These packages should be built as described in their
+corresponding pages linked in the menu on the left.
+
+See the Archer2 [github repository][2] for further details.
```
Comment (aturner-epcc) [inline docs/software-libraries/trilinos.md:58]: Typo, should be ARCHER2

**Review by aturner-epcc — CHANGES_REQUESTED** [review]
Looks mostly fine. Just a few typos for the service name

File: docs/software-libraries/boost.md (hunk 2 above)
Comment (kevinstratford) [reply docs/software-libraries/boost.md]: Fixed (I hope)

**Review by jsindt — DISMISSED** [review]
Typos have been fixed so LGTM

**Review by juanfrh — APPROVED** [review]

**Pull request merged**